File: Prácticas/P1/src/deepspace/WeaponType.java
/**
 * @author elena
 * @brief Representa los tipos de armas del juego y su potencia de disparo.
 */
package deepspace;

public enum WeaponType {
    LASER(2.0f), MISSILE(3.0f), PLASMA(4.0f);
        
    private float power;

    WeaponType(float another_power){
        this.power= another_power;
    }

    float getPower(){
        return power;
    }
    
}

File: Prácticas/ruby/lib/weapon_type.rb
=begin
Representa a los tipos de armas del juego. Cada valor posible del tipo
enumerado tendrá asociado un valor numérico concreto igual a la potencia de disparo de cada tipo
de arma.
=end

module Deepspace
  module WeaponType
     class Type
       def initialize (another_power)
         @power= another_power
       end
       
       attr_reader :power
     end

     LASER= Type.new(2.0)
     MISSILE= Type.new(3.0)
     PLASMA= Type.new(4.0)
  end
end
